LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -191,7 +194,45 @@ public List<Module> GenerateModuleList()
 
         public void CreateClassrooms(int classroomCount)
         {
-            throw new NotImplementedException();
+            foreach(Module module in moduleDictionary.Values)
+            {
+                temporaryModuleHashSet.Add(module);
+
+                Classroom classroom = ObjectGenerator.GenerateClassroom(module);
+
+                classRoomDictionary.Add(Guid.NewGuid(), classroom);
+
+                temporaryModuleHashSet.Remove(module);
+            }
+        }
+
+        public void AssignInstructorsToClassrooms()
```

Comment:
Your logic here is not efficient. You have three nested loops. Think how you can reduce the number of loops by providing an additional data structure. When you assign modules to instructors, there is a way to keep track of which instructors teach which modules.